From ec1e350f0bc185fbe7b2ed1f0fd5ee518c05bb85 Mon Sep 17 00:00:00 2001
From: Han Sol Jin <hansol@hansol.ca>
Date: Fri, 17 May 2024 20:13:26 -0700
Subject: [PATCH 1/3] sepolicy: isolate power HAL policies into a new
 subdirectory

This makes power HAL policies opt-in. Primary reason to do this is to
avoid breaking Pixel builds, which have several conflicting policies.

Change-Id: I66197a7145d424984acf31fe07d282d6e5bcdc8b
---
 libperfmgr/sepolicy.mk                 | 2 ++
 libperfmgr/vendor/hal_power_default.te | 7 +++++++
 libperfmgr/vendor/property.te          | 2 ++
 libperfmgr/vendor/property_contexts    | 2 ++
 libperfmgr/vendor/vendor_init.te       | 2 ++
 5 files changed, 15 insertions(+)
 create mode 100644 libperfmgr/sepolicy.mk
 create mode 100644 libperfmgr/vendor/hal_power_default.te
 create mode 100644 libperfmgr/vendor/property.te
 create mode 100644 libperfmgr/vendor/property_contexts
 create mode 100644 libperfmgr/vendor/vendor_init.te

diff --git a/libperfmgr/sepolicy.mk b/libperfmgr/sepolicy.mk
new file mode 100644
index 0000000..46a7497
--- /dev/null
+++ b/libperfmgr/sepolicy.mk
@@ -0,0 +1,2 @@
+BOARD_VENDOR_SEPOLICY_DIRS += \
+    device/lineage/sepolicy/libperfmgr/vendor
diff --git a/libperfmgr/vendor/hal_power_default.te b/libperfmgr/vendor/hal_power_default.te
new file mode 100644
index 0000000..3169237
--- /dev/null
+++ b/libperfmgr/vendor/hal_power_default.te
@@ -0,0 +1,7 @@
+# To do powerhint on nodes defined in powerhint.json
+allow hal_power_default cgroup:dir search;
+allow hal_power_default cgroup:file rw_file_perms;
+allow hal_power_default sysfs_devices_system_cpu:file rw_file_perms;
+
+# To get/set powerhal state property
+set_prop(hal_power_default, vendor_power_prop)
diff --git a/libperfmgr/vendor/property.te b/libperfmgr/vendor/property.te
new file mode 100644
index 0000000..df317c6
--- /dev/null
+++ b/libperfmgr/vendor/property.te
@@ -0,0 +1,2 @@
+# Power HAL
+vendor_public_prop(vendor_power_prop);
diff --git a/libperfmgr/vendor/property_contexts b/libperfmgr/vendor/property_contexts
new file mode 100644
index 0000000..e37ced2
--- /dev/null
+++ b/libperfmgr/vendor/property_contexts
@@ -0,0 +1,2 @@
+# Power HAL
+vendor.powerhal.                   u:object_r:vendor_power_prop:s0
diff --git a/libperfmgr/vendor/vendor_init.te b/libperfmgr/vendor/vendor_init.te
new file mode 100644
index 0000000..f48eff3
--- /dev/null
+++ b/libperfmgr/vendor/vendor_init.te
@@ -0,0 +1,2 @@
+# To set powerhal init property
+set_prop(vendor_init, vendor_power_prop)
-- 
2.53.0

